### DISCUSSION TITLE
Get a Certificate for imap into DMS

### DISCUSSION THREAD

**Discussion by l0ggik** [docker-mailserver/docker-mailserver#4030] [Q&A]
Hi,

it might just be me being blind but i am not sure how to solve the following problem because i can't find any informations about this:

I have setup DMS from the instructions, everything works perfectly fine and i am really pleased with the result.

I also created a Cerificate for mail.mydomain.com and mounted it into the DMS Container as a custom certificate.

But when i try to connect with a mail client with imap i get an error because the certificate is for mail.mydomain.com and not imap.mydomain.com

I can create a certificate for imap.mydomain.com but how do i correctly mount a second certificate into DMS. The certificate folder for mail.mydomain.com ist mounted to /tmp/dms/custom-certs/ so i can't directly mount a second folder to this folder.
Can i mount a second certificate to /tmp/dms/custom-certs/imap.mydomain.com/ ? Would DMS be able to recognise this as the correct certificate for imap.mydomain.com?

Or how would be the correct way to mount a second certificate into the DMS container?

Help would be really appreciated.

**Comment by glime-ai** [comment]
**Please note** that this response was created using generative AI, leveraging previously available information from the repository. This information may be helpful. However, it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Hi,

To handle multiple certificates for different subdomains in Docker Mailserver (DMS), you can indeed use separate directories within your mounted volume for each certificate. Here’s how you can do it:

1. **Create Certificates**: Ensure you have valid SSL certificates for both `mail.mydomain.com` and `imap.mydomain.com`. You can use Let's Encrypt or any other certificate authority to generate these.

2. **Prepare the Directory Structure on Your Host**: On your host machine, organize the certificates in separate directories within the same parent directory. For example:
   ```
   /path/to/your/certs/mail.mydomain.com/
   /path/to/your/certs/imap.mydomain.com/
   ```

3. **Mount the Certificates in DMS**: Modify your `docker-compose.yml` file to mount the parent directory to a path inside the container. For example:
   ```yaml
   volumes:
     - /path/to/your/certs/:/tmp/dms/custom-certs/
   ```

4. **Configure DMS to Use the Correct Certificates**: You need to configure DMS to use the correct certificate based on the domain being accessed. This can typically be done by configuring the mail server software (like Postfix and Dovecot) to look for certificates in the respective directories based on the requested domain.

   You might need to adjust the configuration files for Postfix and Dovecot to specify the path for the certificates. For example, in Dovecot, you might set:
   ```ini
   ssl_cert = </tmp/dms/custom-certs/imap.mydomain.com/fullchain.pem
   ssl_key = </tmp/dms/custom-certs/imap.mydomain.com/privkey.pem
   ```

   And similarly for Postfix.

5. **Restart DMS**: After making these changes, restart your Docker Mailserver to apply the new configuration:
   ```bash
   docker-compose down
   docker-compose up -d
   ```

This setup allows DMS to use the appropriate certificate based on the subdomain being accessed. Make sure that your DNS records for `mail.mydomain.com` and `imap.mydomain.com` are correctly pointing to your server's IP address.

If you encounter any issues with the certificates not being recognized correctly, ensure that the paths in the configuration files are correct and that the Docker volume mounts are properly set up.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=0d5b86fb7eea4a54acac405212c29aba&utm_medium=github).

**Comment by casperklein** [comment]
It's not possible at the moment to use more than one certificate afaik. You may try a SAN certificate.
Or you could just use "mail.mydomain.com" instead of "imap.mydomain.com" when configuring the IMAP client.

**Comment by polarathene** [comment]
- You can provision certificates with more than one FQDN, or you can provision with a wildcard. Either of those should resolve the issue.
- As suggested, have your mail client connect to a different subdomain than it is trying by default.
